compiler/rustc_hir_typeck/src/fallback.rs

```diff
@@ -8,12 +8,17 @@ use rustc_data_structures::{
 use rustc_hir as hir;
 use rustc_hir::intravisit::Visitor;
 use rustc_hir::HirId;
-use rustc_infer::infer::{DefineOpaqueTypes, InferOk};
 use rustc_middle::bug;
+use rustc_infer::{
+    infer::{DefineOpaqueTypes, InferOk},
+    traits::ObligationCause,
+};
```

Comment:
nit: Can you split this up so it's not nested? Just two `use` lines?

Also, you can probably import `ObligationCause` from the same `use` statement as `ObligationCtxt` below.